### DISCUSSION TITLE
SMA4Notes.txt

### DISCUSSION THREAD

**Discussion by city41** [city41/smaghetti#66] [Ideas]
this doc has lots of things in it to grab: https://github.com/city41/smaghetti/blob/main/reference/SMA4Notes.txt

I was able to get [giant question block](https://github.com/city41/smaghetti/wiki/Giant-World#question-blocks) thanks to this doc

The start of giant question block is in smaghetti, under unfinished.

**Comment by city41** [comment]
this doc says `AF` is angry sun (bank 0 and graphic set 2,0,0,0,0,0) but it just stays in place, doesn't attack, and doesn't hurt mario if he touches. I know there is one e-reader level with an angry sun, so I'll check to see how it's defined there.

**Reply by nintyalex** [reply]
I'm pretty sure the Angry Sun only appears in the main game (in Worlds 2-Quicksand and 8-2), where its swoops down (and becomes harmful) once you pass a certain point.

**Reply by city41** [reply]
I could have swore I saw it in another level where you have to jump across small platforms while it's attacking, but you can avoid the fight altogether by going in a pipe and taking an alternate route. But who knows, that might be from another game entirely.

I'm going to add angry sun to unfinished. I have a feeling this really is the sprite for it. But there is probably a second sprite that tells angry sun to start attacking once mario crosses it. If that is true, and we find that sprite, we could probably add angry sun to e-reader levels.

**Reply by nintyalex** [reply]
Sorry, that's World 8-2 of the main game you're referring to.

As for the sprites, I don't know, either. I checked the NES version with SMB3 Workshop, but didn't find anything.

**Reply by city41** [reply]
ah dang, ok. well I added it to unfinished anyway. Worst case scenario it just never graduates to being finished. It's now deployed.

**Comment by nintyalex** [comment]
I've updated and reorganized my SMA4 Notes zip a bit.
[SMA4 Notes.zip](https://github.com/city41/smaghetti/files/6612326/SMA4.Notes.zip)

**Reply by city41** [reply]
oh damn, did you write the doc I was referring to? wow, small world.

**Reply by nintyalex** [reply]
Nope, just edited it.
